backend/src/main/kotlin/io/zell/zdb/log/LogHelper.kt

```diff
@@ -0,0 +1,37 @@
+package io.zell.zdb.log
+
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedEventImpl
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedEventRegistry
+import io.camunda.zeebe.logstreams.impl.log.LoggedEventImpl
+import io.camunda.zeebe.protocol.Protocol
+import io.camunda.zeebe.protocol.impl.record.RecordMetadata
+import io.camunda.zeebe.util.ReflectUtil
+
+
+/**
+ * Reads the given log event details and converts it to a TypedEvent, which can be further processed.
+ *
+ * Based on the value type the typed event is created. If the value type doesn't exist, e.g. if a new version
+ * is read by zdb then a TypedEvent without a record value is returned.
+ *
+ */
+fun convertToTypedEvent(
+    loggedEvent: LoggedEventImpl,
+    metadata: RecordMetadata
+): TypedEventImpl {
+    val typedEvent = TypedEventImpl(Protocol.decodePartitionId(loggedEvent.key))
+
+    val recordValueClass = TypedEventRegistry.EVENT_REGISTRY.get(metadata.getValueType())
+    if (recordValueClass == null) {
+        // it is likely that the value type is a new type, which is not yet supported by zdb
+        typedEvent.wrap(loggedEvent, metadata, null)
```

Comment:
Really great Idea because I realized we don't even need to use the event registry! Since it is msgpack we can directly convert it to json ! Thanks @korthout I will change it when I have time

Example: `MsgPackConverter.convertToJson(UnsafeBuffer(loggedEvent.valueBuffer, loggedEvent.valueOffset, loggedEvent.valueLength))`

Reply:
@Zelldon I think it would be a good improvement, but the current proposed solution already improves things a lot as well. IMO you can merge it first and look at this further improvement later